src/MeshKernel/Mesh.cpp

```diff
@@ -637,22 +637,22 @@ void Mesh::BuildTree(MeshLocations meshLocation)
     }
```

Comment:
in the function "buildtree", should you not check m_nodeRTreeRequiresUpdate before building the tree? now it could e the tree is not empty, but it still requires an update, and in that case it is not built. same for the edge tree, face tree.